lib/hotel_manager.rb

```diff
@@ -0,0 +1,69 @@
+require 'date'
+require 'pry'
+
+module Booking
+  class Hotel_Manager
+    attr_accessor :all_reservations, :all_rooms, :available_rooms, :all_blocks
+
+    def initialize(max_rooms)
+      if max_rooms > 20
+        raise ArgumentError, "Hotel can have 20 rooms max"
+      else
+        @all_rooms = [*1..max_rooms]
+      end
+      @all_reservations = []
+      @available_rooms = []
+      @all_blocks = [] 
+    end
+
+    
+    def add_reservation(checkin_date, checkout_date, room: nil, cost: 200)
+      @all_reservations ||= []
+      res_array = []
+      if room == nil
+        reservation = Booking::Reservation.new(check_availability(checkin_date, checkout_date).first, checkin_date, checkout_date)
+      elsif room != nil
```

Comment:
Hm, why not make this an `if ... else` block instead of an `if ... elsif` block?